src/view_page.cpp

```diff
@@ -873,52 +871,47 @@ void View::DrawBarLine(
         }
     }
 
-    if (form == BARRENDITION_single) {
-        DrawVerticalSegmentedLine(dc, x, line, barLineWidth);
-    }
-    else if (form == BARRENDITION_dashed) {
-        DrawVerticalSegmentedLine(dc, x, line, barLineWidth, dashLength);
-    }
-    else if (form == BARRENDITION_dotted) {
-        DrawVerticalSegmentedLine(dc, x, line, barLineWidth, dotLength);
-    }
-    else if (form == BARRENDITION_rptend) {
-        DrawVerticalSegmentedLine(dc, x, line, barLineWidth);
-        DrawVerticalSegmentedLine(dc, x2 + barLinesGap / 2, line, barLineThickWidth);
-    }
-    else if (form == BARRENDITION_rptboth) {
-        DrawVerticalSegmentedLine(dc, x, line, barLineWidth);
-        DrawVerticalSegmentedLine(dc, x2 + barLinesGap / 2, line, barLineThickWidth);
-        DrawVerticalSegmentedLine(dc, x2 + barLineSeparation + barLinesGap, line, barLineWidth);
-    }
-    else if (form == BARRENDITION_rptstart) {
-        DrawVerticalSegmentedLine(dc, x, line, barLineThickWidth);
-        DrawVerticalSegmentedLine(dc, x2 + barLineThickWidth / 2, line, barLineWidth);
-    }
-    else if (form == BARRENDITION_invis) {
-        barLine->SetEmptyBB();
-    }
-    else if (form == BARRENDITION_end) {
-        DrawVerticalSegmentedLine(dc, x, line, barLineWidth);
-        DrawVerticalSegmentedLine(dc, x2 + barLinesGap / 2, line, barLineThickWidth);
-    }
-    else if (form == BARRENDITION_dbl) {
-        // Narrow the bars a little bit - should be centered?
-        DrawVerticalSegmentedLine(dc, x, line, barLineWidth);
-        DrawVerticalSegmentedLine(dc, x2 + barLineWidth, line, barLineWidth);
-    }
-    else if (form == BARRENDITION_dbldashed) {
-        DrawVerticalSegmentedLine(dc, x, line, barLineWidth, dashLength);
-        DrawVerticalSegmentedLine(dc, x2 + barLineWidth, line, barLineWidth, dashLength);
-    }
-    else if (form == BARRENDITION_dbldotted) {
-        DrawVerticalSegmentedLine(dc, x, line, barLineWidth, dotLength);
-        DrawVerticalSegmentedLine(dc, x2, line, barLineWidth, dotLength);
-    }
-    else {
-        // Use solid barline as fallback
-        LogWarning("%s bar lines not supported", barLine->AttBarLineLog::BarrenditionToStr(barLine->GetForm()).c_str());
-        DrawVerticalSegmentedLine(dc, x, line, barLineWidth);
+    switch (form) {
```

Comment:
May I ask: what is the reason for changing to a `switch`?

Reply:
I find it more pleasant to read. Also we have this in many other places when it comes to data types.

Reply:
I find it less pleasant, but that's not important.